File: src/main/java/io/zeitwert/ddd/doc/model/enums/CodeCaseStageEnum.java

package io.zeitwert.ddd.doc.model.enums;

import org.springframework.stereotype.Component;

import io.zeitwert.ddd.app.service.api.impl.Enumerations;
import io.zeitwert.ddd.enums.model.base.EnumerationBase;

@Component("codeCaseStageEnum")
public class CodeCaseStageEnum extends EnumerationBase<CodeCaseStage> {

	static private CodeCaseStageEnum INSTANCE;

	protected CodeCaseStageEnum(Enumerations enums) {
		super(CodeCaseStage.class, enums);
		INSTANCE = this;
	}

	public static CodeCaseStageEnum getInstance() {
		return INSTANCE;
	}

	public void addItem(CodeCaseStage item) {
		super.addItem(item);
	}

	public static CodeCaseStage getCaseStage(String caseStageId) {
		return INSTANCE.getItem(caseStageId);
	}

}

File: fm-dddrive/src/main/java/dddrive/app/doc/model/enums/CodeCaseStageEnum.kt
package dddrive.app.doc.model.enums

import dddrive.ddd.model.base.EnumerationBase

class CodeCaseStageEnum : EnumerationBase<CodeCaseStage>(CodeCaseStage::class.java) {

	init {
		instance = this
	}

	companion object {

		lateinit var instance: CodeCaseStageEnum

		@JvmStatic
		fun getCaseStage(caseStageId: String): CodeCaseStage = instance.getItem(caseStageId)

	}

}
